dashboard/app/models/sections/omni_auth_section.rb

```diff
@@ -28,11 +28,13 @@
 
 class OmniAuthSection < Section
   def self.from_omniauth(code:, type:, owner_id:, students:, section_name: 'New Section')
-    oauth_section = with_deleted.where(code: code).first_or_create! do |section|
-      section.name = section_name || 'New Section'
-      section.user_id = owner_id
-      section.login_type = type
-    end
+    oauth_section = with_deleted.where(code: code).first_or_create
+
+    oauth_section.name = section_name || 'New Section'
+    oauth_section.user_id = owner_id
+    oauth_section.login_type = type
+
+    oauth_section.save!
```

Comment:
Nit: do we want to only save if changed?  (We also have this on line 46/48 below.)

```ruby
oauth_section.save! if oauth_section.changed?
```

Reply:
I'll give this a shot - I think it shouldn't break anything, in which case I'm happy to include it to reduce potentially unnecessary saves.